File: RetrofitRxJavaDagger2/app/src/main/java/com/calvert/retrofitrxjavadagger2/di/MyComponent.java
package com.calvert.retrofitrxjavadagger2.di;

import com.calvert.retrofitrxjavadagger2.MainActivity;

import javax.inject.Singleton;

import dagger.Component;
import dagger.Module;

@Singleton
@Component(modules = {RetrofitModule.class})
public interface MyComponent {
    void injectRetrofit(MainActivity mainActivity);
}

File: WeatherKotlinRetrofitRxJavaDagger/app/src/main/java/com/calvert/weatherkotlinretrofitrxjavadagger/di/MyComponent.kt
package com.calvert.weatherkotlinretrofitrxjavadagger.di

import com.calvert.weatherkotlinretrofitrxjavadagger.MainActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(RetrofitModule::class))
interface MyComponent {
    fun injectRetrofit(mainActivity: MainActivity)
}